scripts/refcodes_with_properties/EntryPropertyCalculator.py

```diff
@@ -0,0 +1,582 @@
+#
+# This script can be used for any purpose without limitation subject to the
+# conditions at http://www.ccdc.cam.ac.uk/Community/Pages/Licences/v2.aspx
+#
+# This permission notice and the following statement of attribution must be
+# included in all copies or substantial portions of this script.
+#
+# 2025-03-14: created by Jason C. Cole, The Cambridge Crystallographic Data Centre
+
+
+'''
+Utility classes for filtering CSD entries based on a property control file
+'''
+
+import ccdc.io
+
+_filter_classes = {}
+def register(cls):
+    ''' Register a filter class to use in the script. 
+    :param cls: the class to register
+    '''
+    if cls.name() in _filter_classes:
+        raise ValueError(f"a class with the name {cls.name()} is already registered. Use a different name")
+
+    _filter_classes[cls.name()] = cls
+
+def filter(name):
+   return _filter_classes[name]
+
+def helptext():
+    ''' Get help text
+    '''
+    txt = ""
+    for name in _filter_classes.keys():
+       cls = _filter_classes[name]
+       txt = txt + "    %s -> %s," % ( name, cls.helptext() )
+    return txt[:-1]
+
+
+class _Filter(object):
+
+    @staticmethod
+    def name():
+        raise NotImplementedError # override this
+
+    @staticmethod
+    def helptext():
+        raise NotImplementedError # override this
+
+    @staticmethod
+    def argument_pair():
+        raise NotImplementedError # override this
+
+class _ComparativeFilter(_Filter):
+    def __init__(self, args):
+        value = False
+        if eval(args.strip()) == 1:
+           value = True
+
+        self.expected_value = value
+
+    def value(self):
+        raise NotImplementedError # override this
+
+    def __call__(self,theobject):
+        value = self.value(theobject)
+        return value == self.expected_value
+
+
+class _RangeFilter(_Filter):
+    def __init__(self, args):
+
+        parts = [ p.strip() for p in args.split() ]
+        self.minimum = eval(parts[0])
+        self.maximum = eval(parts[1])
```

Comment:
Same points about eval. is this a string that needs converting to a float? wrapping it in `float(parts[0])`  should be enough